pkg/pillar/agentlog/loglevel.go

```diff
@@ -69,16 +71,18 @@ func getRemoteLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetRemoteLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
+	gc := cast.CastConfigItemValueMap(m)
 	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.RemoteLogLevel != "" {
-		return as.RemoteLogLevel, true
+	as, ok := gc.AgentSettings[agentName]["remote.loglevel"]
+
+	if ok && as.StringValue != "" {
+		return as.Key, true
 	}
 	// Do we have a default value?
-	if allowDefault && gc.DefaultRemoteLogLevel != "" {
+	// TODO: UNSURE HOW TO GET THIS WITH NEW GLOBAL STRUCT
+	/*if allowDefault && gc.DefaultRemoteLogLevel != "" {
```

Comment:
Same comment as above..For remote.loglevel:

1) If per-agent remote.loglevel is set - use it.
2) ELSE - if Global remote loglevel is set - Use it
3) ELSE - if per-agent loglevel is set - use it
4) ELSE - if Global loglevel is set - use it.

(4) --> This is always set - so you will always have the last resort..

Reply:
@kalyan-nidumolu Incorrect.
The remote and (local) loglevels are separate.
Hence for remote look for a per-agent, and if not found use the gobal remote.
And for non-remote look for a per-agent non-remote, and if not found use the global non-remote.

Reply:
I updated the code to reflect this. I think it's correct